hamiltonian_generative_network.py

```diff
@@ -55,25 +56,27 @@ def __init__(self,
         self.optimizer = optimizer
         self.loss = loss
 
-    def forward(self, rollout_batch, n_steps=None):
+    def forward(self, rollout_batch, n_steps=None, variational=True):
         """Get the prediction of the HGN for a given rollout_batch of n_steps.
 
         Args:
-            rollout_batch (torch.Tensor(N, C, H, W)): Image sequence of the system evolution concatenated along the channels' axis.
-            n_steps (integer, optional): Number of guessed steps, if None will match seq_len. Defaults to None.
+            rollout_batch (torch.Tensor): Minibatch of rollouts as a Tensor of shape
+                (batch_size, seq_len, channels, height, width).
+            n_steps (integer, optional): Number of guessed steps, if None will match seq_len.
+                Defaults to None.
+            variational (bool): Whether to sample from the encoder distribution or take the mean.
 
         Returns:
-            HgnResult: Bundle of the intermediate and final results of the HGN output.
+            An HgnResult object containing data of the forward pass over the given minibatch.
```

Comment:
```suggestion
            (utilities.HgnResult): An HgnResult object containing data of the forward pass over the given minibatch.
```

Why don't you like the
```
returns:
(type): Bla bla
```
format? I would use it for consistency or if there is a good reason to use it, change all the others as well.

Reply:
That's actually not a bad idea. I had changed it because most of our code was not using it and do not often see the `Returns` section in that format. But we can use it, it's actually quite clear.